src/main/takeDetection/__tests__/takeDetection.test.ts

```diff
@@ -0,0 +1,331 @@
+import { AssertionError } from 'assert';
+import { Word } from 'sharedTypes';
+import { InjectableTakeGroup } from 'main/editDelete/injectTakeInfo';
+import { findSentences, Sentence, findTakes } from '../takeDetection';
+import getSimilarityScore from '../sentenceSimilarity';
+
+function mockGetSentenceSimilarity(
+  sentenceOne: string,
+  sentenceTwo: string
+): number {
+  if (sentenceOne === 'Hello world!' && sentenceTwo === 'Hi there.') {
+    return 0.8;
+  }
+  if (sentenceOne === 'Hi there.' && sentenceTwo === 'All ok?') {
+    return 0.2;
+  }
+  if (sentenceOne === 'All ok?' && sentenceTwo === 'test.') {
+    return 0.4;
+  }
+  if (sentenceOne === 'test.' && sentenceTwo === 'assessment.') {
+    return 0.8;
+  }
+  if (sentenceOne === 'assessment.' && sentenceTwo === 'exam') {
+    return 0.8;
+  }
+  throw AssertionError;
+}
+
+jest.mock('../sentenceSimilarity');
+const mockedSentenceSim = getSimilarityScore as jest.Mocked<
+  typeof getSimilarityScore
+>;
+
+describe('findSentences should return a list of Sentence objects based on punctuation and findTakes should correctly identify different takes', () => {
+  it('should produce expected sentences when given a list of words', () => {
+    const wordList: Word[] = [
+      {
```

Comment:
There's a `makeBasicWord` util for this where you can just override the things you like and the others will be sane defaults. E.g. you're not modifying the startTime or anything so it's redundant to keep defining it